tests/test_all_dichotomies.py

```diff
@@ -0,0 +1,26 @@
+import pytest
+
+from tests.conftest import trained_decoder
+
+
+@pytest.mark.filterwarnings("ignore::sklearn.exceptions.ConvergenceWarning")
+@pytest.mark.parametrize("trained_decoder", [("base", "base"),
+                                              ("base", "svc"),
+                                             ("base", "nonlinear")
+                                              ], indirect=["trained_decoder"])
+def test_variable_dichotomies(trained_decoder) -> None:
```

Comment:
These are called "semantic" dichotomies around the project.

Reply:
These are now referred to as `linearly_separable_semantic_dichotomies`  and `non_linearly_separable_semantic_dichotomies`. The variable name is quite long. Given it's not user-exposed and IDE's have autocomplete, I figured it best to be explicit.